families/supplychain/python/sawtooth_supplychain/rest_api/route_handlers.py

```diff
@@ -0,0 +1,848 @@
+                      "AND application.id <= %s "
+                      "ORDER BY application.id limit %s")
+            query_tuple += (p_min, p_max, count)
+        else:
+            query += (" AND application.id >= %s "
+                      "ORDER BY application.id limit %s")
+            query_tuple += (p_min, count)
+
+        # Get a connection and cursor
+        try:
+            conn = self._db_connect()
+        except psycopg2.OperationalError:
+            LOGGER.debug("Could not connect to database.")
+            raise errors.DatabaseConnectionError()
+
+        cur = conn.cursor()
+
+        # Execute the query
+        try:
+            cur.execute(query, query_tuple)
+        except:
+            LOGGER.debug("Could not execute query: %s", query)
+            raise errors.UnknownDatabaseError()
+
+        # Fetch the rows
+        rows = cur.fetchall()
+
+        fields = ('identifier', 'applicant', 'type', 'status', 'terms')
+        data = self._make_dict(fields, rows)
+
+        return self._wrap_response(
+            request,
+            data=data,
+            metadata=""
+        )
+
+    @staticmethod
+    def _make_dict(fields, rows):
+        """
+        Convert returned data in the form of lists from DB to dicts,
+        and insert the fields as keys.
+        """
+
+        data = []
+
+        for row in rows:
+            record = {}
+            for i, f in enumerate(fields):
+                if f in ['type', 'status']:
+                    record[f] = row[i].strip()
+                else:
+                    record[f] = row[i]
+            data.append(record)
+        return data
+
+    @staticmethod
+    def add_cors_headers(request, headers):
+        if 'Origin' in request.headers:
+            headers['Access-Control-Allow-Origin'] = request.headers['Origin']
+            headers["Access-Control-Allow-Methods"] = "GET,POST"
+            headers["Access-Control-Allow-Headers"] =\
+                "Origin, X-Requested-With, Content-Type, Accept"
+
+    @staticmethod
+    def _wrap_response(request, data=None, metadata=None, status=200):
+        """Creates the JSON response envelope to be sent back to the client.
+        """
+        envelope = metadata or {}
+
+        if data is not None:
+            envelope['data'] = data
+
+        headers = {}
+        RouteHandler.add_cors_headers(request, headers)
+
+        return web.Response(
+            status=status,
+            content_type='application/json',
+            headers=headers,
+            text=json.dumps(
+                envelope,
+                indent=2,
+                separators=(',', ': '),
+                sort_keys=True))
+
+    @classmethod
+    def _wrap_paginated_response(cls, request, response, data):
+        """Builds the metadata for a pagingated response and wraps everying in
+        a JSON encoded web.Response
+        """
+        # head = response['head_id']
+        link = cls._build_url(request, head=head)
```

Comment:
As your specs do not include `head` or `link` properties in the response, these two lines should be removed (not commented out).